CLI script to gather environment info

BramVanroy opened the pull request:
I noticed that all too often people leave the "Environment" section in their issue empty. However, things such as the version number of PT/TF and `transformers` itself are very useful to know when trying to debug things.

This PR adds a small script to the existing CLI workflow. Running `python transformers-cli info` will output something like this:

```
Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 2.4.0
- Platform: Windows-10-10.0.18362-SP0
- Python version: 3.6.8
- PyTorch version (GPU?): 1.4.0 (True)
- Tensorflow version (GPU?): 2.1.0 (True)
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

Note that GPUs being available through the DL framework (`GPU?`) are retrieved automatically, but that users still have to specify whether or not they are actually using the GPU.

In addition, the relevant issue templates have been updated to direct users to the script.

LysandreJik commented:
This is really cool!

BramVanroy commented:
> This is really cool!

Props to spaCy, since I basically stole [the idea](https://github.com/explosion/spaCy/blob/master/spacy/cli/info.py) from them.

julien-c commented:
LGTM but I've also pinged @mfuntowicz as he will have good insight

BramVanroy commented:
Is there a way to see which test are run in `check_code_quality`? I'm curious as to why it fails.

LysandreJik commented:
It fails because the file `/home/circleci/transformers/src/transformers/commands/info.py` would be reformatted by black.

You can run `make style` at the root to set everything to black style.

BramVanroy commented:
> It fails because the file `/home/circleci/transformers/src/transformers/commands/info.py` would be reformatted by black.
> 
> You can run `make style` at the root to set everything to black style.

Thanks. Is that different from running `black .`? I did that, and it formats all files (not only info.py).

LysandreJik commented:
The black command we used uses a specific line-length which is different to the default (we use a line length of 119, we like it better). We also set it to be based on Python 3.5.

I believe the test now fails because of isort; that's weird, it should have been triggered by the `make style` as well and should have fixed the imports on its own.

BramVanroy commented:
> The black command we used uses a specific line-length which is different to the default (we use a line length of 119, we like it better). We also set it to be based on Python 3.5.
> 
> I believe the test now fails because of isort; that's weird, it should have been triggered by the `make style` as well and should have fixed the imports on its own.

I am now manually running black/isort (without the 'check' flag) and pushing those commits in hopes that the tests will then pass. But, correct me if I'm wrong, isn't circleCI supposed to apply these runs (black, isort etc) before running the tests?

julien-c commented:
Did you install isort with the exact version that's pinned in `CONTRIBUTING.md`?

If you do, both `make style` and `make quality` should reliably pass.

mfuntowicz commented on src/transformers/commands/info.py:
```diff
@@ -0,0 +1,61 @@
+                # returns list of devices, convert to bool
+                tf_cuda_available = bool(tf.config.list_physical_devices("GPU"))
+        except ImportError:
+            pass
```
_Suggestion_ -> rely on transformers methods to check platform availability:

```python
from transformers import is_torch_available, is_tf_available

if is_torch_available():
...

if is_tf_available():
...
```

BramVanroy replied:
Thanks, that make sense. Changed in https://github.com/huggingface/transformers/pull/2699/commits/f0d5aded94b2e6e80ca31ace8fc6ce82aba75ab9

mfuntowicz commented:
It's a very cool feature to provide through the CLI.

I may suggest to rename the command from `info` to `env` as we may want to keep `info` for exposing information about models through cards / config.

What do you think ? @BramVanroy @julien-c @LysandreJik

BramVanroy commented:
> Did you install isort with the exact version that's pinned in `CONTRIBUTING.md`?
> 
> If you do, both `make style` and `make quality` should reliably pass.

Ah, I missed the note on isort. Fixed it now.


> It's a very cool feature to provide through the CLI.
> 
> I may suggest to rename the command from `info` to `env` as we may want to keep `info` for exposing information about models through cards / config.
> 
> What do you think ? @BramVanroy @julien-c @LysandreJik

Good suggestion and future-proof! I renamed the CLI method to env, so the full command is 

```
python transformers-cli env
```

Issue templates have been update and this command has been added to CONTRIBUTING.md as well.

thomwolf commented:
LGTM, thanks a lot @BramVanroy
